### PR TITLE
Adds cell timer logs

### PR DESCRIPTION
**Pull request by Fruerlund**
Sorry for all the github spam
OLD PR: https://github.com/ParadiseSS13/Paradise/pull/7228

Upon starting a cell timer user will now be prompted for names and charges. If both are given it will print a cell log(paper) at prison management consoles which can be stored by the Warden. 

Afterwards the timer will start and door close. If you don't fill out both inputs the cell timer will not run. If you choose to adjust time during sentence it will not ask for inputs again or print a new paper.

UPDATE. These cell logs will also be saved to the record computer and can be printed again by choosing it from the "Print Cell Log" Button. Other way around, these cell logs can also be deleted from "Record Maintenance"

example
example
example
example

Thanks to all the help and handholding from everyone! 

:cl: Fruerlund
add: adds a new game mechanic. When activating a cell timer it will demand input and charges then print a paper at prison management consoles
add: adds saving cell logs and the option to delete them
edit: Edits a lowercase to uppercase in the template.
:cl:

### CODE REVIEW

**Comment by Fruerlund** [comment]
@Crazylemon64 fixed the leading merge commits

**Comment by Crazylemon64** [comment]
Thanks!

**Comment by Bram961** [comment]
I feel like this is a good feature all the way up to the point where it says "the cell timer will not run" I feel like that should be left up to the players, not to mention how sometimes you just throw people in for 2 minutes when they start throwing a drunken tantrum in the bar.

**Comment by FalseIncarnate** [comment]
Anyone who is getting brigged has to have a valid reason under Space Law for being brigged. If they threw a "drunken tantrum" that resulted in them getting brigged, it was likely they were arrested for battery at the MINIMUM. If an officer can't be bothered listing that they were brigged for that reason, then that officer probably shouldn't be brigging anyone.

Further, this immensely alleviates the ever-present issue of poor communication between sec members. I can't tell you how many times I've gotten ahelps from prisoners who don't know why they were brigged and then the warden or HoS has to ask around for 5 minutes just to find out why. Requiring them to insert a reason (doesn't even have to be a good one, but does have to be something) makes it much that easier to investigate that since I can track down a piece of paper on a console much faster than playing the telephone game with a member of sec to get a reason.

**Comment by deep-fried-faceplant** [comment]
As a long time security player, I really like this. However, I want to see pictures of the UI before I give this the IAA stamp of approval.

**Comment by FalseIncarnate** [comment]
The UI is literally 2 popups that ask for the name and the reason. Just like when you try and rename a paper or something like that.

**Comment by Fruerlund** [comment]
At the moment there is no UI, which is fine.. In the future I might push an update that adds these records to the security records computer.

**Comment by Fruerlund** [comment]
UPDATE Logs will now be stored in a list and a button has been added to the security records console. This button will allow users to select a log and print that log and the security record console. If no logs are available, users will be met with "No files available."

At the moment logs will still be automatically spawned at prison management consoles every time a cell timer is activated as well as be stored in the record computer. I am not sure if I should keep both or one or the other.

**Comment by Birdtalon** [comment]
👍 This is a good QoL feature and definitely should be merged.

**Comment by scrubmcnoob** [comment]
Really looking forward to this if it does get merged in.

File: code/game/machinery/computer/security.dm [hunk 1]
```diff
@@ -342,6 +342,18 @@
 					create_record_photo(active1)
 				printing = 0
 
+		else if(href_list["printlogs"])
+			var/obj/item/weapon/paper/P = (input(usr, "Select log to print", "Available Cell Logs") as null|anything in cell_logs)
+			if(!P)
+				return 0
+			playsound(src.loc, "sound/goonstation/machines/printer_dotmatrix.ogg", 50, 1)
```
Comment (tigercat2000) [inline code/game/machinery/computer/security.dm]: No `src.` please, it is implied.

File: code/game/machinery/computer/security.dm [hunk 2]
```diff
@@ -342,6 +342,18 @@
...
 				printing = 0
 
+		else if(href_list["printlogs"])
+			var/obj/item/weapon/paper/P = (input(usr, "Select log to print", "Available Cell Logs") as null|anything in cell_logs)
+			if(!P)
+				return 0
+			playsound(src.loc, "sound/goonstation/machines/printer_dotmatrix.ogg", 50, 1)
+			to_chat(usr, "<span class='danger'>Printing [P.name].</span>")
+			sleep(50)
+			var/obj/item/weapon/paper/LOG = new /obj/item/weapon/paper(src.loc)
```
Comment (tigercat2000) [inline code/game/machinery/computer/security.dm]: No `src.` please, it is implied.

Comment (tigercat2000) [inline code/game/machinery/computer/security.dm]: Also, minor note, but `LOG` would be better `log` for naming consistency.

File: code/game/machinery/computer/security.dm [hunk 3]
```diff
@@ -354,6 +366,7 @@
 			var/index = min(max(text2num(href_list["del_c"]) + 1, 1), length(active2.fields["comments"]))
 			if(istype(active2, /datum/data/record) && active2.fields["comments"][index])
 				active2.fields["comments"] -= active2.fields["comments"][index]
+				return 0
```
Comment (tigercat2000) [inline code/game/machinery/computer/security.dm]: Is there a reason this is needed?

File: code/game/machinery/doors/brigdoors.dm [hunk 4]
```diff
@@ -277,6 +309,7 @@
 				F.flash()
 
 		if(href_list["change"])
+			printed = 1
```
Comment (tigercat2000) [inline code/game/machinery/doors/brigdoors.dm]: Why is this here, rather than in `timer_start`?

File: code/game/machinery/computer/security.dm [hunk 5]
```diff
@@ -342,6 +342,18 @@
 					create_record_photo(active1)
 				printing = 0
 
+		else if(href_list["printlogs"])
+			var/obj/item/weapon/paper/P = (input(usr, "Select log to print", "Available Cell Logs") as null|anything in cell_logs)
```
Comment (tigercat2000) [inline code/game/machinery/computer/security.dm]: The extra set of brackets around `(input(` are unnecessary.

File: code/game/machinery/doors/brigdoors.dm (hunk 4 above)
Comment (Fruerlund) [inline code/game/machinery/doors/brigdoors.dm]: The reason there is a printed = 1, is because when you adjust time and hit change, it will run the timer_start again, but with the new time. This means that users would be prompted for input and charges and print a whole new paper everytime the time is adjusted. This printed = 1, stops the print proc from running.

**Comment by Fruerlund** [comment]
@tigercat2000 
Update: Added the possibility to delete cell logs. All changes should also be sorted.
